rosbag2_transport/src/rosbag2_transport/play_options_from_node_params.cpp

```diff
@@ -0,0 +1,132 @@
+  auto read_ahead_queue_size_ = node->declare_parameter<int64_t>(
+    "read_ahead_queue_size",
+    1000,
+    desc_raqs);
+  play_options.read_ahead_queue_size = static_cast<uint64_t>(read_ahead_queue_size_);
+
+  play_options.node_prefix = node->declare_parameter<std::string>(
+    "node_prefix",
+    "");
+
+  auto desc_rate = param_utils::float_param_description(
+    "Playback rate (hz)",
+    0.000001,
+    std::numeric_limits<float>::max());
+  play_options.rate = node->declare_parameter<float>(
+    "rate",
+    1.0,
+    desc_rate);
+
+  play_options.topics_to_filter = node->declare_parameter<std::vector<std::string>>(
+    "topics_to_filter",
+    std::vector<std::string>());
+
+  play_options.topics_regex_to_filter = node->declare_parameter<std::string>(
+    "topics_regex_to_filter",
+    "");
+
+  play_options.topics_regex_to_exclude = node->declare_parameter<std::string>(
+    "topics_regex_to_exclude",
+    "");
+
+  // TODO(roncapat)
+  // std::unordered_map<std::string, rclcpp::QoS> topic_qos_profile_overrides = {};
+
+  play_options.loop = node->declare_parameter<bool>(
+    "loop",
+    false);
+
+  auto topic_remapping_options = node->declare_parameter<std::vector<std::string>>(
+    "topic_remapping_options",
+    std::vector<std::string>());
+
+  if (!topic_remapping_options.empty()) {
+    RCLCPP_WARN(
+      node->get_logger(),
+      "Remappings shall be applied through standard CLI/Launch options."
+      "'topic_remapping_options' content will be ignored.");
+  }
+
+  play_options.clock_publish_frequency = node->declare_parameter<double>(
+    "clock_publish_frequency",
+    0.0);
+
+  play_options.clock_publish_on_topic_publish = node->declare_parameter<bool>(
+    "clock_publish_on_topic_publish",
+    false);
+
+  play_options.clock_trigger_topics = node->declare_parameter<std::vector<std::string>>(
+    "clock_trigger_topics",
+    std::vector<std::string>());
+
+  auto playback_duration_ = node->declare_parameter<double>(
+    "playback_duration",
+    -1);
+  play_options.playback_duration = rclcpp::Duration::from_seconds(playback_duration_);
+
+  play_options.playback_until_timestamp = node->declare_parameter<int64_t>(
+    "playback_until_timestamp",
+    -1);
+
+  play_options.start_paused = node->declare_parameter<bool>(
+    "start_paused",
+    false);
+
+  play_options.start_offset = node->declare_parameter<int64_t>(
+    "start_offset",
+    0);
+
+  play_options.disable_keyboard_controls = node->declare_parameter<bool>(
+    "disable_keyboard_controls",
+    false);
+
+  // TODO(roncapat): but these are never set in ros2 bag play verb.
+  // KeyboardHandler::KeyCode pause_resume_toggle_key = KeyboardHandler::KeyCode::SPACE;
+  // KeyboardHandler::KeyCode play_next_key = KeyboardHandler::KeyCode::CURSOR_RIGHT;
+  // KeyboardHandler::KeyCode increase_rate_key = KeyboardHandler::KeyCode::CURSOR_UP;
+  // KeyboardHandler::KeyCode decrease_rate_key = KeyboardHandler::KeyCode::CURSOR_DOWN;
+
```

Comment:
Please remove this TODO with keyboard handler key codes initialization. 
There is no way to pass key binding constants via node params any way.